consensus/src/epoch_manager.rs

```diff
@@ -1512,6 +1513,44 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
         Ok(())
     }
 
+    fn check_author(&mut self, peer_id: AccountAddress, msg: &ConsensusMsg) -> anyhow::Result<()> {
+        let author = match msg {
+            ConsensusMsg::CommitMessage(commit) => commit.author(),
+            ConsensusMsg::ProposalMsg(proposal) => proposal.proposal().author(),
+            ConsensusMsg::VoteMsg(vote) => Some(vote.vote().author()),
+            ConsensusMsg::OrderVoteMsg(order_vote) => Some(order_vote.order_vote().author()),
+            ConsensusMsg::CommitVoteMsg(commit_vote) => Some(commit_vote.author()),
+            ConsensusMsg::BatchMsg(batch) => batch.author(),
+            ConsensusMsg::RoundTimeoutMsg(round_timeout) => Some(round_timeout.author()),
+            ConsensusMsg::BatchResponse(batch_response) => Some(batch_response.author()),
+            ConsensusMsg::BatchRequestMsg(batch_request) => Some(batch_request.source()),
+            ConsensusMsg::SignedBatchInfo(sign_batch_info) => sign_batch_info.author(),
```

Comment:
it also feels awkward to only check the first author, probably better to have this check to individual verify function instead of here

Reply:
Yes. The verify function already checks the author [here](https://github.com/aptos-labs/aptos-core/blob/satya/econia-tests/consensus/consensus-types/src/proof_of_store.rs#L264) for each individual signed batch info. So, I originally ignored the author check here by setting author to `None`. Should I rever it back?

Reply:
I mean instead of having this giant check_author function, check the author inside each individual message's verify function or the UnverifiedEvent::verify function as Balaji mentioned above

Reply:
Moved the sender verification checks to individual verify functions.